### PR TITLE
James/bug fixes

### PR DESCRIPTION
**Pull request by voidSevenSevenSix**
Bug fixes, filled placeholders

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/arm/Arm.java [hunk 1]
```diff
@@ -87,34 +84,28 @@ public void startScore() {
 
   /** True if the arm is at the queried position. */
   public boolean atPosition(ArmPosition positionToCheck) {
-    double armMarginDegrees =
-        positionToCheck == ArmPosition.STARTING
-            ? ArmCal.ARM_START_MARGIN_DEGREES
-            : ArmCal.ARM_MARGIN_DEGREES;
+    double armMarginDegrees = ArmCal.ARM_MARGIN_DEGREES;
```
Comment (balghane) [inline src/main/java/frc/robot/subsystems/arm/Arm.java:87]: Don't need this local variable

File: src/main/java/frc/robot/subsystems/arm/Arm.java [hunk 2]
```diff
@@ -87,34 +84,28 @@ public void startScore() {
...
     return Math.abs(armPositionDegrees - armPositionToCheckDegrees) <= armMarginDegrees;
   }
 
-  /** Returns the cosine of the arm angle in degrees off of the horizontal. */
-  public double getCosineArmAngle() {
-    return Math.cos(
-        Units.degreesToRadians(
-            armEncoder.getPosition() - ArmConstants.ARM_POSITION_WHEN_HORIZONTAL_DEGREES));
+  /** Returns the arm angle in degrees off of the horizontal. */
+  public double getArmAngle() {
```
Comment (balghane) [inline src/main/java/frc/robot/subsystems/arm/Arm.java:96]: Add `RelativeToHorizontal` to function name to make it clear

File: src/main/java/frc/robot/subsystems/arm/Arm.java [hunk 3]
```diff
@@ -178,27 +169,21 @@ public static void setDegreesFromGearRatioRelativeEncoder(
 
   /** True if the arm is at the queried position. */
   public boolean atDesiredArmPosition() {
-    double armMarginDegrees =
-        desiredPosition == ArmPosition.STARTING
-            ? ArmCal.ARM_START_MARGIN_DEGREES
-            : ArmCal.ARM_MARGIN_DEGREES;
+    double armMarginDegrees = ArmCal.ARM_MARGIN_DEGREES;
```
Comment (balghane) [inline src/main/java/frc/robot/subsystems/arm/Arm.java:172]: Also don't need this here

File: src/main/java/frc/robot/subsystems/grabber/Grabber.java [hunk 4]
```diff
@@ -18,16 +21,26 @@ public class Grabber extends SubsystemBase {
   private final DigitalInput gamePieceSensor =
       new DigitalInput(RobotMap.GRABBER_GAME_PIECE_SENSOR_DIO);
 
+  private boolean runningCommand = false;
+
   public void spinMotors(double power) {
+    runningCommand = power==0.0?false:true;
```
Comment (balghane) [inline src/main/java/frc/robot/subsystems/grabber/Grabber.java:27]: Checking floating point equality with zero is bad practice generally, maybe just check that abs(power) is less than some small very number

File: src/main/java/frc/robot/subsystems/grabber/Grabber.java [hunk 5]
```diff
@@ -43,6 +56,13 @@ public boolean seeGamePiece() {
     return !gamePieceSensor.get();
   }
 
+  public void initSparks(){
+    frontMotor.restoreFactoryDefaults();
+    backMotor.restoreFactoryDefaults();
+    backMotor.follow(frontMotor);
+    frontMotor.setSmartCurrentLimit(GrabberCalibrations.MOTOR_CURRENT_LIMIT);
```
Comment (balghane) [inline src/main/java/frc/robot/subsystems/grabber/Grabber.java:63]: Set for both

File: src/main/java/frc/robot/subsystems/grabber/Grabber.java [hunk 6]
```diff
@@ -51,5 +71,9 @@ public void initSendable(SendableBuilder builder) {
     builder.addBooleanProperty("Sensor sees game piece", this::seeGamePiece, null);
   }
 
-  public void periodic() {}
+  public void periodic() {
+    if(seeGamePiece() && !runningCommand){
+      spinMotors(0.1);
```
Comment (balghane) [inline src/main/java/frc/robot/subsystems/grabber/Grabber.java:76]: Add 0.1 to calibrations

File: src/main/java/frc/robot/subsystems/arm/Arm.java [hunk 7]
```diff
@@ -178,27 +169,21 @@ public static void setDegreesFromGearRatioRelativeEncoder(
...
+    
   }
 
-  /** Cancellation function */
-  public void cancelScore() {
-    cancelScore = true;
+  /**Cancellation function */
+  public void cancelScore(){
+      goToPosition(ArmPosition.STARTING);
+      isScoring = false;
```
Comment (balghane) [inline src/main/java/frc/robot/subsystems/arm/Arm.java:186]: Is the plan for the FinishScore to check isScoring or is the plan to check isScoring then not run FinishScore at all if it's false?

File: src/main/java/frc/robot/subsystems/grabber/Grabber.java [hunk 8]
```diff
@@ -24,7 +24,7 @@ public class Grabber extends SubsystemBase {
   private boolean runningCommand = false;
 
   public void spinMotors(double power) {
-    runningCommand = power==0.0?false:true;
+    runningCommand = Math.abs(power)<0.01?false:true;
```
Comment (tails618) [inline src/main/java/frc/robot/subsystems/grabber/Grabber.java:27]: Return `!condition`

**Review by tails618 — APPROVED** [review]

**Pull request merged**